static/js/ratings.js

```diff
@@ -0,0 +1,39 @@
+async function getRating() {
+    const id = window.location.href.slice(window.location.href.lastIndexOf("/") + 1);
+    
+    const res = await fetch(`/api/movies/${id}/rating`);
+    const data = await res.json();
+    const rating = data.rating;
+
+    const container = document.querySelector("#rating-container-element");
+    const ul = document.querySelector(".movie-rating-hats");
+    
+    if (rating !== "API Error") {
+    
+        for (let i=0; i<5; i++) {
+            
+            let li = document.createElement("li");
+            
+            if (i < rating) {
+            
+                if ( i == Math.floor(rating)) {
+            
+                    li.classList.add("hat-halffill");
+            
+                } else {
+            
+                    li.classList.add("hat-fill")
+                }
+            } else {
+
+                li.classList.add("hat-nofill")
+            }
+
+            ul.appendChild(li);
+        }
+        
+        container.append(ul);
```

Comment:
I think you forgot to declare the ul variable outside of the for loop, I'm gettning a reference error.

Reply:
You already fixed it